src/adapter/graphql/src/queries/flows/flow.rs

```diff
@@ -28,60 +30,121 @@ use crate::utils;
 #[derive(Clone)]
 pub struct Flow {
     flow_state: Box<fs::FlowState>,
+    description: FlowDescription,
 }
 
 #[Object]
 impl Flow {
     #[graphql(skip)]
-    pub fn new(flow_state: fs::FlowState) -> Self {
-        Self {
-            flow_state: Box::new(flow_state),
+    pub async fn build_batch(
+        flow_states: Vec<fs::FlowState>,
+        ctx: &Context<'_>,
+    ) -> Result<Vec<Self>> {
+        let mut result: Vec<Self> = Vec::new();
+        // We need this HashMap to avoid multiple queries to the same dataset polling
+        // source and cover cases when dataset has no Ingest flows, so we will
+        // build flow descriptions without searching of polling sources
+        //
+        // In addition it might be useful if we will add another entity which cause
+        // duplicate requests
+        let mut dataset_polling_sources: HashMap<
+            opendatafabric::DatasetID,
+            Option<(
+                odf::Multihash,
+                odf::MetadataBlockTyped<odf::SetPollingSource>,
+            )>,
+        > = HashMap::new();
+
+        for flow_state in &flow_states {
+            let flow_description =
+                Self::build_description(ctx, flow_state, &mut dataset_polling_sources).await?;
+            result.push(Self {
+                flow_state: Box::new(flow_state.clone()),
```

Comment:
Why do we need this clone? We own the states: `flow_states: Vec<fs::FlowState>`.
As we don't use them later, this clone feels unneeded.